static/script/devices/mediaplayer/samsung_streaming.js

```diff
@@ -465,49 +469,51 @@ require.def(
                     this._emitEvent(MediaPlayer.EVENT.STATUS);
                 }
             },
-
-            _onMetadata: function() {
-                this.logger.info('Calling this.playerPlugin.Execute("GetPlayingRange")');
-                var range = this._playerPlugin.Execute("GetPlayingRange").split('-');
-                this._range = {
-                    start: range[0],
-                    end: range[1]
-                };
-
-                this.logger.info('Current Duration ' + this._range.end * 1000 + 'ms');
-
-                // this.logger.info("_onMetadata range = {0, "  + this._range.end);
-                // this.logger.info("this._currentTime = " + this._currentTime);       //<TODO> so currentTime is currently in milliseconds relative to midnight! and not to hls window
-
-                if (this._waitingToPlayFrom()) {
-                    this._deferredPlayFrom();
-                }
-
-                this._onFinishedBuffering();
-            },
-
-            _waitingToPlayFrom: function() {
-                return this._targetSeekTime !== undefined;
-            },
-
-            _deferredPlayFrom: function() {
-                this._seekTo(this._targetSeekTime);
-
-                if (this._postBufferingState === MediaPlayer.STATE.PAUSED) {
-                    this._tryPauseWithStateTransition();
+            
+            _updateRange: function () {
+                var self = this;
+                if (this._currentPlayer === this.PlayerEmps.StreamingPlayer) {
+                    var range = this._playerPlugin.Execute("GetPlayingRange").split('-');
+                    this._range = {
+                        start: Math.floor(range[0]),
+                        end: Math.floor(range[1])
+                    };
+                    //don't call range for the next 8 seconds
+                    this._updatingTime = true;
```

Comment:
As mentioned elsewhere in this review, it would probably be best to update range every 8 seconds of real time, with this function initially being triggered by the `STREAM_INFO_READY` event. We would also be able to remove `_updatingTime`.